### PR TITLE
Add table aws_inspector_assessment_template. closes #244

### LINKED ISSUE #244: Add table aws_inspector_assessment_template
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/inspector

### PR DESCRIPTION
**Pull request by sayan133**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_inspector_assessment_template []

PRETEST: tests/aws_inspector_assessment_template

TEST: tests/aws_inspector_assessment_template
Running terraform
aws_inspector_resource_group.inspector_rg: Creating...
aws_inspector_resource_group.inspector_rg: Creation complete after 3s [id=arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-lJQcHkJg]
aws_inspector_assessment_target.assessment_target: Creating...
aws_inspector_assessment_target.assessment_target: Creation complete after 2s [id=arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB]
aws_inspector_assessment_template.named_test_resource: Creating...
aws_inspector_assessment_template.named_test_resource: Creation complete after 5s [id=arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB/template/0-snECtLVF]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = "388460667113"
assessment_target_aka = "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB/template/0-snECtLVF"
resource_name = "turbottest10976"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "388460667113",
    "arn": "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB/template/0-snECtLVF",
    "assessment_run_count": 0,
    "assessment_target_arn": "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB",
    "duration_in_seconds": 3600,
    "last_assessment_run_arn": "<null>",
    "name": "turbottest10976",
    "partition": "aws",
    "region": "us-east-1",
    "tags_src": [
      {
        "Key": "foo",
        "Value": "bar"
      }
    ],
    "user_attributes_for_findings": []
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB/template/0-snECtLVF",
    "assessment_target_arn": "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB",
    "duration_in_seconds": 3600,
    "name": "turbottest10976"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:inspector:us-east-1:388460667113:target/0-dHes6XFB/template/0-snECtLVF"
    ],
    "tags": {
      "foo": "bar"
    },
    "title": "turbottest10976"
  }
]
✔ PASSED

POSTTEST: tests/aws_inspector_assessment_template

TEARDOWN: tests/aws_inspector_assessment_template

SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  name,
  arn,
  assessment_run_count,
  created_at,
  assessment_target_arn,
  region
from
  aws_inspector_assessment_template;
```
```
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------
| name          | arn                                                                            | assessment_run_count | created_at          | assessment_target_arn             
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------
| new-test      | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-9I6Xgqdu | 0                    | 2021-03-23 12:39:31 | arn:aws:inspector:us-east-1:388460
| test-stempipe | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-4iQokEzn | 0                    | 2021-03-23 11:42:19 | arn:aws:inspector:us-east-1:388460
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------

```
### List Assessment Templates that have no user attributes for findings tags

```sql
select
  name,
  arn,
  assessment_run_count,
  created_at,
  assessment_target_arn,
  user_attributes_for_findings,
  region
from
  aws_inspector_assessment_template
where
  user_attributes_for_findings = '[]';
```
```
+----------+--------------------------------------------------------------------------------+----------------------+---------------------+----------------------------------------
| name     | arn                                                                            | assessment_run_count | created_at          | assessment_target_arn                  
+----------+--------------------------------------------------------------------------------+----------------------+---------------------+----------------------------------------
| new-test | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-9I6Xgqdu | 0                    | 2021-03-23 12:39:31 | arn:aws:inspector:us-east-1:38846066711
+----------+--------------------------------------------------------------------------------+----------------------+---------------------+----------------------------------------

```

### List Assessment Templates that have zero assessment runs

```sql
select
  name,
  arn,
  assessment_run_count,
  created_at,
  assessment_target_arn,
  user_attributes_for_findings,
  region
from
  aws_inspector_assessment_template
where
  assessment_run_count = 0;
```
```
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------
| name          | arn                                                                            | assessment_run_count | created_at          | assessment_target_arn             
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------
| new-test      | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-9I6Xgqdu | 0                    | 2021-03-23 12:39:31 | arn:aws:inspector:us-east-1:388460
| test-stempipe | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-4iQokEzn | 0                    | 2021-03-23 11:42:19 | arn:aws:inspector:us-east-1:388460
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------

```
### List Assessment Templates with assessment run duration less than 1 hour

```sql
select
  name,
  arn,
  assessment_run_count,
  created_at,
  assessment_target_arn,
  duration_in_seconds,
  region
from
  aws_inspector_assessment_template
where
  duration_in_seconds < 3600;
```
```
+------+-----+----------------------+------------+-----------------------+---------------------+--------+
| name | arn | assessment_run_count | created_at | assessment_target_arn | duration_in_seconds | region |
+------+-----+----------------------+------------+-----------------------+---------------------+--------+
+------+-----+----------------------+------------+-----------------------+---------------------+--------+

```
### List Assessment Templates that are created within last 7 days

```sql
select
  name,
  arn,
  assessment_run_count,
  created_at,
  assessment_target_arn,
  region
from
  aws_inspector_assessment_template
where
  created_at > (current_date - interval '7' day);
```
```
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------
| name          | arn                                                                            | assessment_run_count | created_at          | assessment_target_arn             
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------
| new-test      | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-9I6Xgqdu | 0                    | 2021-03-23 12:39:31 | arn:aws:inspector:us-east-1:388460
| test-stempipe | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE/template/0-4iQokEzn | 0                    | 2021-03-23 11:42:19 | arn:aws:inspector:us-east-1:388460
+---------------+--------------------------------------------------------------------------------+----------------------+---------------------+-----------------------------------

```
```
</details>

### CODE REVIEW

File: aws-test/tests/aws_inspector_assessment_template/variables.tf [hunk 1]
```diff
@@ -0,0 +1,95 @@
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "integration-tests"
```
Comment (khushboo9024) [inline aws-test/tests/aws_inspector_assessment_template/variables.tf:9]:
```suggestion
  default     = "default"
```

File: aws/table_aws_inspector_assessment_template.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+				Transform:   transform.FromField("Arn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAssessmentTemplates(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_template.go:118]:
```suggestion
```

File: aws/table_aws_inspector_assessment_template.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+				Type:        pb.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAssessmentTemplates(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_template.go:117]:
```suggestion
func listInspectorAssessmentTemplates(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_inspector_assessment_template.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func describeAssessmentTemplate(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_template.go:150]:
```suggestion
func getInspectorAssessmentTemplate(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_inspector_assessment_template.go [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+	logger := plugin.Logger(ctx)
+	logger.Trace("describeAssessmentTemplate")
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_template.go:154]:
```suggestion
```

File: aws/table_aws_inspector_assessment_template.go [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+		AssessmentTemplateArns: []*string{aws.String(assessmentTemplateArn)},
+	}
+
+	// Get call
+	data, err := svc.DescribeAssessmentTemplates(params)
+	if err != nil {
+		logger.Debug("describeAssessmentTemplate__", "ERROR", err)
+		return nil, err
+	}
+	if len(data.AssessmentTemplates) > 0 {
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_template.go:185]:
```suggestion
	if data.AssessmentTargets != nil && len(data.AssessmentTemplates) > 0 {
```

File: aws/table_aws_inspector_assessment_template.go [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+	}
+	return nil, nil
+}
+
+// API call for fetching tag list
+func getAwsInspectorAssessmentTemplateTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsInspectorAssessmentTemplateTags")
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_template.go:196]:
```suggestion
```

File: docs/tables/aws_inspector_assessment_template.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  duration_in_seconds,
+  region
+from
+  aws_inspector_assessment_template
+where
+  duration_in_seconds < 3600;
+```
+
+
+### List assessment templates that are created within last 7 days
```
Comment (Paulami30) [inline docs/tables/aws_inspector_assessment_template.md:76]:
```suggestion
### List assessment templates that are created before last 7 days
```

File: docs/tables/aws_inspector_assessment_template.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  assessment_run_count,
+  created_at,
+  assessment_target_arn,
+  region
+from
+  aws_inspector_assessment_template;
+```
+
+
+### List assessment templates that have no user attributes for findings tags
```
Comment (Paulami30) [inline docs/tables/aws_inspector_assessment_template.md:22]: please add a meaningful heading

**Review by Paulami30 — COMMENTED** [review]
resolve merge conflict

**Review by Paulami30 — APPROVED** [review]

**Comment by Paulami30** [comment]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**